pkg/icingadb/runtime_updates.go

```diff
@@ -107,13 +202,26 @@ func (r *RuntimeUpdates) Sync(ctx context.Context, factoryFuncs []contracts.Enti
 			}
 		})
 
-		deleted := make(chan interface{})
 		g.Go(func() error {
-			defer close(deleted)
+			defer func(ch chan interface{}) {
+				select {
+				case <-ch:
+				default:
+					close(ch)
+				}
```

Comment:
This operation is only safe if there's no concurrent write/close action on the same channel.